libcxx/docs/ReleaseNotes/19.rst

```diff
@@ -55,6 +55,10 @@ Improvements and New Features
 Deprecations and Removals
 -------------------------
 
+- The C++20 synchronization library (``<barrier>``, ``<latch>``, ``atomic::wait``, etc.) has been deprecated
+  in language modes prior to C++20. If you are using these features prior to C++20, please update to ``-std=c++20``.
+  In LLVM 20, the C++20 synchronization library will be removed entirely in language modes prior to C++20.
```

Comment:
Don't we typically deprecate for 2 releases?

Reply:
I don't think we do? Or at least I don't think we have any sort of fixed policy for doing that?

I'm pretty sure we've deprecated stuff in one release and removed it in the next in some cases, and we might also have kept stuff deprecated for 2 release before removing it. Or is my memory playing tricks on me and we've actually always deprecated for 2 releases before removing? CC @philnik777

Reply:
I checked and it indeed seems we use a shorter deprecation (nowadays?). So I'm oke to keep this as is.